tests/units/classes/test.php

```diff
@@ -1245,5 +1250,51 @@ public function testStaticMethod()
 						->isEqualTo(array($return1, $return2, $return3))
 			;
 		}
+
+		public function testNewMockInstance()
+		{
+			$this
+				->if($test = new emptyTest())
+				->then
+					->object($mock = $test->newMockInstance('stdClass'))
+						->isInstanceOf('mock\stdClass')
+						->isInstanceOf('stdClass')
+					->object($test->newMockInstance('stdClass'))
+						->isInstanceOf('mock\stdClass')
+						->isInstanceOf('stdClass')
+						->isNotIdenticalTo($mock)
+					->object($test->newMockInstance('stdClass', 'foobar'))
+						->isInstanceOf('foobar\stdClass')
+						->isInstanceOf('stdClass')
+					->object($test->newMockInstance('stdClass', 'foo', 'bar'))
+						->isInstanceOf('foo\bar')
+						->isInstanceOf('stdClass')
+
+				->given($arguments = array($firstArgument = uniqid(), $secondArgument = rand(0, PHP_INT_MAX)))
```

Comment:
Here we check that when we pass constructor arguments they are correctly passed to the mock constructor